evm/src/Transceiver/WormholeTransceiver/WormholeTransceiverState.sol

```diff
@@ -0,0 +1,280 @@
+import "../../interfaces/INttManager.sol";
+
+import "../Transceiver.sol";
+
+abstract contract WormholeTransceiverState is IWormholeTransceiverState, Transceiver {
+    using BytesParsing for bytes;
+
+    // ==================== Immutables ===============================================
+    uint8 public immutable consistencyLevel;
+    IWormhole public immutable wormhole;
+    IWormholeRelayer public immutable wormholeRelayer;
+    ISpecialRelayer public immutable specialRelayer;
+    uint256 public immutable wormholeTransceiver_evmChainId;
+
+    // ==================== Constants ================================================
+    uint256 public constant GAS_LIMIT = 500000;
+
+    /// @dev Prefix for all TransceiverMessage payloads
+    ///      This is 0x99'E''W''H'
+    /// @notice Magic string (constant value set by messaging provider) that idenfies the payload as an transceiver-emitted payload.
+    ///         Note that this is not a security critical field. It's meant to be used by messaging providers to identify which messages are Transceiver-related.
+    bytes4 constant WH_TRANSCEIVER_PAYLOAD_PREFIX = 0x9945FF10;
+
+    /// @dev Prefix for all Wormhole transceiver initialisation payloads
+    ///      This is bytes4(keccak256("WormholeTransceiverInit"))
+    bytes4 constant WH_TRANSCEIVER_INIT_PREFIX = 0x9c23bd3b;
+
+    /// @dev Prefix for all Wormhole peer registration payloads
+    ///      This is bytes4(keccak256("WormholePeerRegistration"))
+    bytes4 constant WH_PEER_REGISTRATION_PREFIX = 0x18fc67c2;
+
+    constructor(
+        address nttManager,
+        address wormholeCoreBridge,
+        address wormholeRelayerAddr,
+        address specialRelayerAddr,
+        uint8 _consistencyLevel
+    ) Transceiver(nttManager) {
+        wormhole = IWormhole(wormholeCoreBridge);
+        wormholeRelayer = IWormholeRelayer(wormholeRelayerAddr);
+        specialRelayer = ISpecialRelayer(specialRelayerAddr);
+        wormholeTransceiver_evmChainId = block.chainid;
+        consistencyLevel = _consistencyLevel;
+    }
+
+    enum RelayingType {
+        Standard,
+        Special,
+        Manual
+    }
+
+    function _initialize() internal override {
+        super._initialize();
+        _initializeTransceiver();
+    }
+
+    function _initializeTransceiver() internal {
+        TransceiverStructs.TransceiverInit memory init = TransceiverStructs.TransceiverInit({
+            transceiverIdentifier: WH_TRANSCEIVER_INIT_PREFIX,
+            nttManagerAddress: toWormholeFormat(nttManager),
+            nttManagerMode: INttManager(nttManager).getMode(),
+            tokenAddress: toWormholeFormat(nttManagerToken),
+            tokenDecimals: INttManager(nttManager).tokenDecimals()
+        });
+        wormhole.publishMessage(0, TransceiverStructs.encodeTransceiverInit(init), consistencyLevel);
+    }
+
+    function _checkImmutables() internal view override {
+        assert(this.nttManager() == nttManager);
```

Comment:
Also can you call super._checkImmutables() instead of asserting here, seems cleaner imo

Reply:
Yep :+1: